Registration/RegisterWithAzure.psm1

```diff
@@ -237,67 +237,175 @@ function Set-AzsRegistration{
+    # Configure Azure Bridge
+    $servicePrincipal = New-ServicePrincipal -RefreshToken $azureAccountInfo.Token.RefreshToken -AzureEnvironmentName $AzureContext.Environment.Name -TenantId $azureAccountInfo.TenantId -PSSession $session
 
-        # Configure Azure Bridge
-        $servicePrincipal = New-ServicePrincipal -RefreshToken $azureAccountInfo.Token.RefreshToken -AzureEnvironmentName $AzureContext.Environment.Name -TenantId $azureAccountInfo.TenantId -PSSession $session
+    # Get registration token
+    $getTokenParams = @{
+        BillingModel                  = $BillingModel
+        MarketplaceSyndicationEnabled = $MarketplaceSyndicationEnabled
+        UsageReportingEnabled         = $UsageReportingEnabled
+        AgreementNumber               = $AgreementNumber
+        MsAssetTag                    = $MsAssetTag
+    }
+    Log-Output "Get-RegistrationToken parameters: $(ConvertTo-Json $getTokenParams)"
+    $registrationToken = Get-RegistrationToken @getTokenParams -Session $session -StampInfo $stampInfo
 
-        # Get registration token
-        $getTokenParams = @{
-            BillingModel                  = $BillingModel
-            MarketplaceSyndicationEnabled = $MarketplaceSyndicationEnabled
-            UsageReportingEnabled         = $UsageReportingEnabled
-            AgreementNumber               = $AgreementNumber
-            MsAssetTag                    = $MsAssetTag
-        }
-        Log-Output "Get-RegistrationToken parameters: $(ConvertTo-Json $getTokenParams)"
-        if (($BillingModel -eq 'Capacity') -and ($UsageReportingEnabled))
-        {
-            Log-Warning "Billing model is set to Capacity and Usage Reporting is enabled. This data will be used to guide product improvements and will not be used for billing purposes. If this is not desired please halt this operation and rerun with -UsageReportingEnabled:`$false. Execution will continue in 20 seconds."
-            Start-Sleep -Seconds 20        
-        }
-        $registrationToken = Get-RegistrationToken @getTokenParams -Session $session -StampInfo $stampInfo
-    
-        # Register environment with Azure
+    $session | Remove-PSSession
+    return @{
+        ServicePrincipalObjectId = $servicePrincipal.ObjectId
+        RegistrationToken        = $registrationToken
+    }
 
-        Log-Output "Creating registration resource at ResourceGroupLocation: $ResourceGroupLocation"
-        New-RegistrationResource -ResourceGroupName $ResourceGroupName -ResourceGroupLocation $ResourceGroupLocation -RegistrationToken $RegistrationToken -RegistrationName $RegistrationName
+}
 
-        # Assign custom RBAC role
-        Log-Output "Assigning custom RBAC role to resource $RegistrationName"
-        New-RBACAssignment -SubscriptionId $AzureContext.Subscription.SubscriptionId -ResourceGroupName $ResourceGroupName -RegistrationName $RegistrationName -ServicePrincipal $servicePrincipal
+Invoke-RegistrationStep2AgainstAzure{
```

Comment:
idk about these names. when user runs this command, it's going to spit out a giant encoded string, and there isn't much telling them what that is. Maybe return an object instead, like: ````return [pscustomobject]@{ ActivationKey = $key }```` as one way to help indicate how this is flowing. Or via the function name, like "Register-AzureStackAndRetrieveActivationKey" or maybe just "Request-AzureStackActivationKey".

Reply:
Maybe something like:
1. ````Begin-AzureStackRegistration````
2. ````Submit-AzureStackRegistration````
3. ````Complete-AzureStackRegistration````

And the wrapper function:
````Invoke-AzureStackRegistration````